consensus/core/src/header.rs

```diff
@@ -88,3 +103,223 @@ impl Header {
         }
     }
 }
+
+#[wasm_bindgen]
+impl Header {
+    /// Finalizes the header and recomputes (updates) the header hash
+    /// @return { String } header hash
+    #[wasm_bindgen(js_name = finalize)]
+    pub fn finalize_js(&mut self) -> String {
+        self.hash = hashing::header::hash(self);
+        self.hash.to_hex()
+    }
+
+    #[wasm_bindgen(constructor)]
+    pub fn constructor(js_value: JsValue) -> std::result::Result<Header, JsError> {
+        Ok(js_value.try_into()?)
+    }
+
+    /// Obtain `JSON` representation of the header. JSON representation
+    /// should be obtained using WASM, to ensure proper serialization of
+    /// big integers.
+    #[wasm_bindgen(js_name = asJSON)]
+    pub fn as_json(&self) -> String {
+        serde_json::to_string(self).unwrap()
+    }
+
+    #[wasm_bindgen(getter = hash)]
+    pub fn get_hash_as_hex(&self) -> String {
+        self.hash.to_hex()
+    }
+
+    #[wasm_bindgen(setter = hash)]
+    pub fn set_hash_from_js_value(&mut self, js_value: JsValue) {
```

Comment:
is there ever a need to set the hash from JS? (instead of finalizing)

Reply:
I don't think there is.  I think this was added by accident (usually you bind properties via getters and setters bi-directionally).  removing.

Reply:
Note that there's a TryFrom below which also sets the hash from a js value